cpp/tests/copying/scatter_list_scalar_tests.cpp

```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/null_mask.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+
+namespace cudf {
+namespace test {
+
+using mask_vector = std::vector<valid_type>;
+using size_column = fixed_width_column_wrapper<size_type>;
+
+class ScatterListScalarTests : public cudf::test::BaseFixture {
+};
+
+void test_single_scalar_scatter(column_view const& target,
```

Comment:
I implemented this way in the past for shortening the code, but found that it is very difficult to debug. Instead, we should use a function that just does some intermediate steps and return the final result (i.e., the function returns scatter result):
```
auto get_scatter(...) 
{
  ...
  return result->column(0);
}
```

Then, you call `CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT` at the call site. By doing so, whenever a test fails you will know exactly what EXPECT line fails.

Reply:
Interesting thoughts. When a test fails, doesn't google-test shows which test case fails?

Reply:
In your case, you have only one assert per test thus you can identify the test case. If you have more than one assertion per test then it's more difficult. Nevertheless, you still need to search for the test name, instead of jumping directly to the line identified by the macro.